sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumer/EventHubConsumerLiveTests.cs

```diff
@@ -1240,5 +1602,127 @@ await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsum
+
+                            Assert.That(async () => await newConsumer.ReceiveAsync(1, TimeSpan.Zero), Throws.Nothing);
+
+                            consumers.Add(newConsumer);
+                        }
+
+                        // Attempt to create 6th consumer. This should fail.
+
+                        var failConsumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroupName, partition, EventPosition.Latest);
+
+                        await failConsumer.ReceiveAsync(1, TimeSpan.Zero);
+
+                        throw new InvalidOperationException("6th consumer should have encountered QuotaExceededException.");
+                    }
+                    catch(TrackOne.QuotaExceededException ex)
+                    {
+                        for (var i = 0; i < 5; i++)
+                        {
+                            var identifier = $"consumer{i}";
+                            Assert.That(ex.Message.Contains(identifier), Is.True, $"QuotaExceededException message is missing consumer identifier '{identifier}')");
+                        }
+                    }
+                    finally
+                    {
+                        await Task.WhenAll(consumers.Select(consumer => consumer.CloseAsync()));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task SmallReceiveWaitTimeDoesNotThrowEventHubsTimeoutException()
+        {
+            // Issue receives with 1 second so that some of the Receive calls will timeout while creating AMQP link.
+            // Even those Receive calls should return NULL instead of bubbling the exception up.
+
+            var receiveTimeoutInSeconds = 1;
+
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    var tasks = Enumerable.Range(0, 10)
+                        .Select(async i =>
+                        {
+                            await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroupName, partition, EventPosition.Latest))
+                            {
+                                Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.FromSeconds(receiveTimeoutInSeconds)), Throws.Nothing);
+                            }
+                        });
+
+                    await Task.WhenAll(tasks);
+                }
+            }
+        }
+
+#if FullNetFx
```

Comment:
Are we sure that this won't run on .NET Core now that the lower level `netstandard` versions are gone?

Reply:
In fact, it does work for Core. I have removed the preprocessor directives.